src/perfEvents_linux.cpp

```diff
@@ -101,16 +106,56 @@ int PerfEvents::tid() {
 }
 
 void PerfEvents::createForThread(int tid) {
+    __u32 type;
+    __u64 config;
+    int precise_ip = 2;
+    switch (_event_type){
+        case EVENT_TYPE_CPU_CLOCK:
+            type = PERF_TYPE_SOFTWARE;
+            config = PERF_COUNT_HW_CPU_CYCLES;
+            break;
+        case EVENT_TYPE_CTX_SWITCHES:
+            type = PERF_TYPE_SOFTWARE;
+            config = PERF_COUNT_SW_CONTEXT_SWITCHES;
+            break;
+        case EVENT_TYPE_BRANCH_MISSES:
+            type = PERF_TYPE_HARDWARE;
+            config = PERF_COUNT_HW_BRANCH_MISSES;
+            break;
+        case EVENT_TYPE_CACHE_MISSES:
+            precise_ip = 0;
+            type = PERF_TYPE_HARDWARE;
+            config = PERF_COUNT_HW_CACHE_MISSES;
+            break;
+        case EVENT_TYPE_CYCLES:
+            type = PERF_TYPE_HARDWARE;
+            config = PERF_COUNT_HW_CPU_CYCLES;
+            break;
+        case EVENT_TYPE_L1D_LOAD_MISSES:
+            precise_ip = 0;
+            type = PERF_TYPE_HW_CACHE;
+            config = hw_cache_event_config(PERF_COUNT_HW_CACHE_L1D,
+                                           PERF_COUNT_HW_CACHE_OP_READ,
+                                           PERF_COUNT_HW_CACHE_RESULT_MISS);
+            break;
+        case EVENT_TYPE_LLC_LOAD_MISSES:
+            precise_ip = 0;
+            type = PERF_TYPE_HW_CACHE;
+            config = hw_cache_event_config(PERF_COUNT_HW_CACHE_LL,
+                                           PERF_COUNT_HW_CACHE_OP_READ,
+                                           PERF_COUNT_HW_CACHE_RESULT_MISS);
+            break;
+    }
```

Comment:
It's already long enough switch, and it'll be even longer when new events added. Probably array-backed mapping will be good.

Reply:
Sounds good